src/main/java/com/google/starfish/services/LabelService.java

```diff
@@ -46,6 +50,57 @@ public void insertMiscLabel(DataSource pool, String label) {
     insertLabel(pool, label, Type.MISC);
   }
 
+  /** Returns a map of all school and course labels in the database */
+  public HashMap<String, String[]> getAllSchoolAndCourseLabels(DataSource pool) {
+    HashMap<String, String[]> organizedLabels = new HashMap<>();
+    try (Connection conn = pool.getConnection()) {
+      List<String> schools = new ArrayList<>();
+      List<String> courses = new ArrayList<>();
+      try {
+        conn.setAutoCommit(false);
+        String stmt1 = 
```

Comment:
nit: since there's more than one statement here, might be more readable if you called them something a little more descriptive like `schoolsStmt` and `coursesStmt` than just using numbers to distinguish.

Reply:
Will do, nice catch!